### PR TITLE
New: Suggested tables

### PR DESCRIPTION
**Pull request by daniellevinson**
## Description

Add suggested sources to source selector (and some styling changes)

## Proposed Changes

New datasource property: `suggestedDataTables: string[]`

## Screenshots

Screen Shot 2021-09-15 at 4 45 02 PM

## License

I confirm that this contribution is made under the terms of the license found in the root directory of this repository's source tree and that I have the authority necessary to make this contribution on behalf of its copyright owner.

### CODE REVIEW

File: packages/reports/addon/components/report-builder/source-selector-list.hbs [hunk 1]
```diff
@@ -0,0 +1,24 @@
+{{!-- Copyright 2021, Yahoo Holdings Inc. Licensed under the terms of the MIT license. See accompanying LICENSE.md file for terms. --}}
+{{#if (gt @sources.length 0)}}
+  <ol class="report-builder-source-selector__list p-l-20">
```
Comment (kevinhinterlong) [inline packages/reports/addon/components/report-builder/source-selector-list.hbs:3]:
```suggestion
  <ol class="report-builder-source-selector__list p-l-20" ...attributes>
```

File: packages/reports/addon/components/report-builder/source-selector.hbs [hunk 2]
```diff
@@ -1,28 +1,53 @@
...
-      </ol>
+              <DenaliIcon
+                @icon="star"
+                @size="extrasmall"
+                class="p-b-3"
+              />
+              Suggested {{@sourceType}}
+            </DenaliChip>
+          </div>
+          <ReportBuilder::SourceSelectorList
```
Comment (kevinhinterlong) [inline packages/reports/addon/components/report-builder/source-selector.hbs:24]: Can we add a class name to distinguish between favorites and all list

File: packages/reports/addon/components/report-builder/source-selector.hbs [hunk 3]
```diff
@@ -1,28 +1,53 @@
...
-                  <EmberTooltip @popperContainer="body" @targetId="{{this.guid}}-{{idx}}" @side="right" @text={{item.description}} />
-                {{/if}}
-              </span>
-              <DenaliIcon class="is-brand-300" @icon="arrowhead-right" />
-            </button>
-          </li>
-        {{/each}}
-      </ol>
+              <DenaliIcon
+                @icon="star"
```
Comment (kevinhinterlong) [inline packages/reports/addon/components/report-builder/source-selector.hbs:17]:
```suggestion
                @icon="star-solid"
```

File: packages/reports/tests/integration/components/report-builder/sidebar-test.ts [hunk 4]
```diff
@@ -123,8 +123,8 @@ module('Integration | Component | report-builder/sidebar', function (hooks) {
 
     assert.deepEqual(
       findAll('.report-builder-source-selector__source-name').map((el) => el.textContent?.trim()),
-      ['Network', 'Protected Table', 'Table A', 'Table B', 'Table C'],
-      'The available tables are listed'
+      ['Network', 'Network', 'Protected Table', 'Table A', 'Table B', 'Table C'],
```
Comment (kevinhinterlong) [inline packages/reports/tests/integration/components/report-builder/sidebar-test.ts:126]: Can we do two assertions for favorites list and all list

File: packages/reports/tests/integration/components/report-builder/source-selector-test.ts [hunk 5]
```diff
@@ -51,33 +55,40 @@ module('Integration | Component | report-builder/source-selector', function (hoo
...
-    });
+      .dom('.report-builder-source-selector-title')
+      .doesNotExist('Titles are not displayed if no sources are suggested.');
 
-    await click('.report-builder-source-selector__source-button[data-source-name="Source B"]');
-    await animationsSettled();
+    assert.deepEqual(
+      findAll('.report-builder-source-selector__source').map((el) => el.textContent?.trim()),
+      ['Source B', 'Source A'],
+      'The sources are listed as passded in'
```
Comment (kevinhinterlong) [inline packages/reports/tests/integration/components/report-builder/source-selector-test.ts:90]:
```suggestion
      'The sources are listed as passed in'
```

**Review by kevinhinterlong — APPROVED** [review]

**Review by cythrawll — APPROVED** [review]

**Pull request merged**